src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectCreationExpression.cs

```diff
@@ -103,31 +103,48 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
 
         public override BoundNode VisitWithExpression(BoundWithExpression withExpr)
         {
-            RoslynDebug.AssertNotNull(withExpr.CloneMethod);
-            Debug.Assert(withExpr.CloneMethod.ParameterCount == 0);
             Debug.Assert(withExpr.Receiver.Type!.Equals(withExpr.Type, TypeCompareKind.ConsiderEverything));
 
             // for a with expression of the form
             //
             //      receiver with { P1 = e1, P2 = e2 }
             //
-            // we want to lower it to a call to the receiver's `Clone` method, then
+            // if the receiver is a struct, then set the given struct properties:
```

Comment:
>given [](start = 57, length = 5)

This sounds somewhat confusing. We are not mutating the given struct, but rather returning a new value.

Reply:
The sentence was incomplete. Fixed

---
In reply to: [607973334](https://github.com/dotnet/roslyn/pull/52319#discussion_r607973334) [](ancestors = 607973334)